gateway/mw_basic_auth.go

```diff
@@ -217,27 +217,33 @@ var errUnauthorized = errors.New("Unauthorized")
 
 func (k *BasicAuthKeyIsValid) checkPassword(session *user.SessionState, plainPassword string, logger *logrus.Entry) error {
 	switch session.BasicAuthData.Hash {
-	case user.HashBCrypt:
-		if err := k.compareHashAndPassword(session.BasicAuthData.Password, plainPassword, logger); err != nil {
-			return err
-		}
 	case user.HashPlainText:
 		if session.BasicAuthData.Password != plainPassword {
 			return errUnauthorized
 		}
-	default:
+
+	case user.HashSha256:
+		fallthrough
+	case user.HashMurmur32:
+		fallthrough
+	case user.HashMurmur64:
+		fallthrough
+	case user.HashMurmur128:
 		// Verify we have a valid Hash value
 		hashAlgo := string(session.BasicAuthData.Hash)
-		basicAuthHashAlgo := storage.HashAlgo(hashAlgo)
-		if basicAuthHashAlgo != hashAlgo {
```

Comment:
@titpetric why this validation is removed?

Reply:
At this point, any hash algo is validated and supported, so HashAlgo() was removed. HashAlgo also had a fallback to murmur32, while we want to keep bcrypt for this behavior.

Reply:
https://github.com/TykTechnologies/tyk/blob/master/gateway/mw_basic_auth.go#L291 - not sure if this is with the right context, but do we really need to have this check here?

Reply:
At first glance, no, but it wasn't the code I touched; Seems like it's comparing a murmur64 hash to a bcrypt hash? Uhhhhhhhhhhhhhhhhhh :dagger:

Reply:
> At this point, any hash algo is validated and supported, so HashAlgo() was removed. HashAlgo also had a fallback to murmur32, while we want to keep bcrypt for this behavior.

All the new/updated Sessions are going to have a hash algo validated/supported. We don't know about the existing ones.  
A misconfiguration there now is going to lead to hashing the password - `storage.HashStr(plainPassword, hashAlgo)` then failing instead of failing before hashing.
It's a super side scenario, but I don't think it would hurt to maintain the check. What do you think?

Reply:
Existing ones already define "bcrypt" as the cipher type (it was not configurable before), so no difference. We keep the default behavior if invalid configuration. A misconfiguration will lead to "bcrypt" as the default, before reaching this code here... api.go:L379 already implements what you're suggesting, I think :)

Reply:
I'm going to resolve this as okay (backwards compatible). The above has already passed QA as well.